### PR TITLE
Bug 1841260: Make resolv.conf prepend operation atomic

### PR DESCRIPTION
**Pull request by cybertron**
This PR includes two changes:

1) It makes resolv.conf prepending atomic. This prevents a potential problem
in case the sed is interrupted, which could leave an empty resolv.conf.

2) It moves the baremetal dispatcher script to common. With the removal of the
DNS VIP, there is no longer any difference between the master and worker
versions of this. This can't be done for the other platforms yet because they
still rely on the DNS VIP.

**- Description for the changelog**
Make resolv.conf prepend operation atomic to prevent potential race conditions.

### CODE REVIEW

File: templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml [hunk 1]
```diff
@@ -25,7 +25,8 @@ contents:
             logger -s "NM resolv-prepender: Prepending 'nameserver $NAMESERVER_IP' to /etc/resolv.conf (other nameservers from /var/run/NetworkManager/resolv.conf)"
             sed -e "/^search/d" \
                 -e "/Generated by/c# Generated by KNI resolv prepender NM dispatcher script\nsearch $DOMAIN\nnameserver $NAMESERVER_IP" \
-                /var/run/NetworkManager/resolv.conf > /etc/resolv.conf
+                /var/run/NetworkManager/resolv.conf > /etc/resolv.tmp
+            mv -f /etc/resolv.tmp /etc/resolv.conf
```
Comment (stbenjam) [inline templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml:29]: There's a sed below at well that limits us to the top 3 nameservers. Should we do it all (including that sed) in a temp file and then copy the file result?

**Comment by stbenjam** [comment]
@cgwalters reccomended using `set -e` and bash strict mode.

Maybe we could also run the script through shellcheck just for good measure?

File: templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml [hunk 2]
```diff
@@ -25,7 +25,8 @@ contents:
...
         else
             logger -s "NM resolv-prepender: Couldn't find a Virtual IP, just updating resolv.conf"
             cp /var/run/NetworkManager/resolv.conf /etc/resolv.conf
```
Comment (cgwalters) [inline templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml:32]: `cp` is also not atomic - this is a really big trap that a whole lot of people have fallen into.  You could probably unify these by having the second path do
`cp /var/run/NetworkManager/resolv.conf /etc/resolv.tmp`
and then running
`mv -f /etc/resolv.tmp /etc/resolv.conf` after the `if`.

File: templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml (hunk 1 above)
Comment (cybertron) [reply templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml:30]: You're right, we should. That fits well with cgwalters suggestion below as well.

File: templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml (hunk 2 above)
Comment (cybertron) [reply templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml:33]: Good idea. Will do.

**Comment by cybertron** [comment]
Hmm, just noticed there is a set +e in here. I'm unsure why we would have done that. I don't see anything that would be expected to fail after that. :-/

**Pull request merged**